Frameworks/Foundation/NSObject_NSKeyValueArrayAdapter.mm

```diff
@@ -132,195 +92,186 @@ bool setIfResponds(id target, const std::string& selectorString, SEL* out) {
 
-        _getMutateAndSet(@selector(removeObjectAtIndex:), index);
+            _getMutateAndSet(@selector(removeObjectAtIndex:), index);
 
-        [_target didChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
+            [_target didChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
+        }
     }
 
     void addObject(id object) {
         if (_targetSelectors.insertAtOne) {
             insertObjectAtIndex(object, count());
-            return;
-        }
-
-        NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:count()];
-        [_target willChange:NSKeyValueChangeInsertion valuesAtIndexes:indexes forKey:_key];
+        } else {
+            NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:count()];
+            [_target willChange:NSKeyValueChangeInsertion valuesAtIndexes:indexes forKey:_key];
 
-        _getMutateAndSet(@selector(addObject:), object);
+            _getMutateAndSet(@selector(addObject:), object);
 
-        [_target didChange:NSKeyValueChangeInsertion valuesAtIndexes:indexes forKey:_key];
+            [_target didChange:NSKeyValueChangeInsertion valuesAtIndexes:indexes forKey:_key];
+        }
     }
 
     void removeLastObject() {
         if (_targetSelectors.removeAtOne) {
             removeObjectAtIndex(count() - 1);
-            return;
-        }
-
-        NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:count() - 1];
-        [_target willChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
+        } else {
+            NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:count() - 1];
+            [_target willChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
 
-        _getMutateAndSet(@selector(removeLastObject));
+            _getMutateAndSet(@selector(removeLastObject));
 
-        [_target didChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
+            [_target didChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
+        }
     }
 
     void replaceObjectAtIndexWithObject(NSUInteger index, id object) {
         if (_targetSelectors.replaceOne) {
-            _call<void>(_targetSelectors.replaceOne, index, object);
-            return;
+            _call<void>(_target, _targetSelectors.replaceOne, index, object);
         } else if (_targetSelectors.removeAtOne && _targetSelectors.insertAtOne) {
-            _call<void>(_targetSelectors.removeAtOne, index);
-            _call<void>(_targetSelectors.insertAtOne, object, index);
-            return;
+            _call<void>(_target, _targetSelectors.removeAtOne, index);
+            _call<void>(_target, _targetSelectors.insertAtOne, object, index);
+        } else {
+            NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:count() - 1];
+            [_target willChange:NSKeyValueChangeReplacement valuesAtIndexes:indexes forKey:_key];
+
+            _getMutateAndSet(@selector(replaceObjectAtIndex:withObject:), index, object);
```

Comment:
This is called on the object we get from _get, and is supposed to fail if it does not exist